src/lib/sql/sql_pipeline_statement.hpp

```diff
@@ -72,6 +72,9 @@ class SQLPipelineStatement : public Noncopyable {
   const std::shared_ptr<SQLPipelineStatementMetrics>& metrics() const;
 
  private:
+  // Checks whether a query plan can be executed without failures
```

Comment:
That wording is too vague, imo. What this function actually does is basic sanity checking before DCL (http://www.datenbanken-verstehen.de/datenbank-grundlagen/dbms/dml-ddl-dcl-kommandos/) operators are executed that would predictably fail - and give the user the option to catch InvalidInputExceptions.

Reply:
I get your point, what about this:

> Performs a sanity check in order to prevent an execution of a predictably failing DDL operator. Throws an InvalidInputException if an invalid LQP is detected.

Reply:
+"DDL operator (e.g. creating a table that already exists")

then, marvelous